### PR TITLE
Feature/gunney/closest point communicator

### PR DESCRIPTION
**Pull request by gunney1**
# Let users set a custom communicator for DistributedClosestPoint

- This PR is a feature
- It does the following (modify list as needed):
  - Add method `DistributedClosestPoint::setMpiCommunicator()`
  - Lets user specify a custom communicator have DistributedClosestPoint duplicates one for private use.

This PR addresses issue https://github.com/LLNL/axom/issues/889

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @gunney1!

With this bugfix, I think (hope) we should be able to run several concurrent calls to the distributed_closest_point example within the same allocation.

Could you also please try removing the logic that I added to the build system that serializes the invocations of the different variants as a workaround for not having a separate comm?

* https://github.com/LLNL/axom/blob/3754c74636641118ac2e77354958a21febde89c5/src/axom/quest/examples/CMakeLists.txt#L55-L57
* https://github.com/LLNL/axom/blob/3754c74636641118ac2e77354958a21febde89c5/src/axom/quest/examples/CMakeLists.txt#L81-L84
* https://github.com/LLNL/axom/blob/3754c74636641118ac2e77354958a21febde89c5/src/axom/quest/examples/CMakeLists.txt#L103-L104

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 1]
```diff
@@ -1411,10 +1468,12 @@ class DistributedClosestPoint
     case 2:
       m_dcp_2->setSquaredDistanceThreshold(m_sqDistanceThreshold);
       m_dcp_2->computeClosestPoints(query_node, cooordset);
+      m_dcp_2->setMpiCommunicator(m_mpiComm);
```
[the remark this answers is not in the document]
Comment (gunney1) [reply src/axom/quest/DistributedClosestPoint.hpp:1471]: Yikes!  Yes it should.

**Comment by gunney1** [comment]
> Thanks @gunney1!
> 
> With this bugfix, I think (hope) we should be able to run several concurrent calls to the distributed_closest_point example within the same allocation.
> 
> Could you also please try removing the logic that I added to the build system that serializes the invocations of the different variants as a workaround for not having a separate comm?
> 
>     * https://github.com/LLNL/axom/blob/3754c74636641118ac2e77354958a21febde89c5/src/axom/quest/examples/CMakeLists.txt#L55-L57
> 
>     * https://github.com/LLNL/axom/blob/3754c74636641118ac2e77354958a21febde89c5/src/axom/quest/examples/CMakeLists.txt#L81-L84
> 
>     * https://github.com/LLNL/axom/blob/3754c74636641118ac2e77354958a21febde89c5/src/axom/quest/examples/CMakeLists.txt#L103-L104

Too many branches modifying the same test, and I was looking at the wrong one.  Sorry.  This is now done.

**Review by agcapps — APPROVED** [review]
Thanks.  This looks good.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**